sklearn/utils/estimator_checks.py

```diff
@@ -847,6 +866,100 @@ def _generate_sparse_matrix(X_csr):
         yield sparse_format + "_64", X
 
 
+def check_array_api_input(
+    name, estimator_orig, *, array_namespace, device=None, dtype="float64"
+):
+    """Check that the array_api Array gives the same results as ndarrays."""
+    try:
+        xp = importlib.import_module(array_namespace)
+    except ModuleNotFoundError:
+        raise SkipTest(
+            f"{array_namespace} is not installed: not checking array_api input"
+        )
+    try:
+        import array_api_compat  # noqa
+    except ImportError:
+        raise SkipTest(
+            "array_api_compat is not installed: not checking array_api input"
+        )
+
```

Comment:
`xp.asarray(..., device=)` does not work for `cupy`. One way to work around it is to use `array_api_compat` to get a unified interface.

```suggestion
    xp = array_api_compat.array_namespace(xp.asarray(1))

```

Reply:
Good point. Added this with a comment explaining why it is needed.